src/bundle/Controller/UserSettingsController.php

```diff
@@ -104,12 +104,24 @@ public function updateAction(Request $request, UpdateView $view)
                     'ibexa_user_settings'
                 );
 
+                $returnUrl = $request->query->has('returnUrl') ? $request->query->get('returnUrl') : null;
+
                 if ($form->getClickedButton() instanceof Button
                     && $form->getClickedButton()->getName() === UserSettingUpdateType::BTN_UPDATE_AND_EDIT
                 ) {
-                    return $this->redirectToRoute('ibexa.user_settings.update', [
+                    $routeParams = [
                         'identifier' => $data->getIdentifier(),
-                    ]);
+                    ];
+                    
+                    if ($returnUrl) {
```

Comment:
Implicit cast to boolean in if condition tends to be error-prone. In this case we're dealing with `?string` (`string|null`) data type, so what you really need here is probably:
```suggestion
                    if (null !== $returnUrl) {
```